homeassistant/components/climate/coolmaster.py

```diff
@@ -0,0 +1,207 @@
+"""
+CoolMasterNet platform that offers control of CoolMasteNet Climate Devices.
+
+For more details about this platform, please refer to the documentation
+https://www.home-assistant.io/components/climate.coolmaster/
+"""
+
+import logging
+
+import voluptuous as vol
+
+from homeassistant.components.climate import (
+    PLATFORM_SCHEMA, STATE_AUTO, STATE_COOL, STATE_DRY, STATE_FAN_ONLY,
+    STATE_HEAT, SUPPORT_FAN_MODE, SUPPORT_ON_OFF, SUPPORT_OPERATION_MODE,
+    SUPPORT_TARGET_TEMPERATURE, ClimateDevice)
+from homeassistant.const import (
+    ATTR_TEMPERATURE, CONF_HOST, CONF_PORT, TEMP_CELSIUS, TEMP_FAHRENHEIT)
+import homeassistant.helpers.config_validation as cv
+
+REQUIREMENTS = ['pycoolmasternet==0.0.4']
+
+SUPPORT_FLAGS = (SUPPORT_TARGET_TEMPERATURE | SUPPORT_FAN_MODE |
+                 SUPPORT_OPERATION_MODE | SUPPORT_ON_OFF)
+
+FAN_ONLY_OVERRIDE = 'fan'
+
+DEFAULT_PORT = 10102
+
+DOMAIN = 'coolmaster'
+
+AVAILABLE_MODES = [STATE_HEAT, STATE_COOL, STATE_AUTO, STATE_DRY,
+                   STATE_FAN_ONLY]
+
+CONF_SUPPORTED_MODES = 'supported_modes'
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_HOST): cv.string,
+    vol.Optional(CONF_PORT, default=DEFAULT_PORT): cv.port,
+    vol.Optional(CONF_SUPPORTED_MODES, default=AVAILABLE_MODES):
+        vol.All(cv.ensure_list, [vol.In(AVAILABLE_MODES)]),
+})
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_platform(hass, config, async_add_entities,
+                               discovery_info=None):
+    """Set up the CoolMasterNet climate platform."""
+    from pycoolmasternet import CoolMasterNet
+
+    supported_modes = config.get(CONF_SUPPORTED_MODES)
+    host = config.get(CONF_HOST)
+    port = config.get(CONF_PORT)
+    cool = CoolMasterNet(host, port=port)
+
+    default_unit = hass.config.units.temperature_unit
+
+    all_devices = [CoolmasterClimate(device, supported_modes, default_unit)
+                   for device in cool.devices()]
+
+    async_add_entities(all_devices, True)
+
+
+class CoolmasterClimate(ClimateDevice):
+    """Representation of a coolmaster climate device."""
+
+    def __init__(self, device, supported_modes, default_unit):
+        """Initialize the climate device."""
+        _LOGGER.debug("Adding device %s", device.uid)
+        self._device = device
+        self._uid = device.uid
+        self._operation_list = supported_modes
+        self._support_flags = SUPPORT_FLAGS
+        self._fan_list = ['low', 'med', 'high', 'auto']
+        self._target_temperature = 0
+        self._current_temperature = 0
+        self._current_fan_mode = ''
+        self._current_operation = ''
+        self._on = False
+        self._unit = default_unit
```

Comment:
I don't think this is needed as initial value should be `None` -- referring to `default_unit`